File: Android/aiconversationkit/src/main/java/com/trtc/uikit/aiconversationkit/manager/net/AIConversationRequest.java
package com.trtc.uikit.aiconversationkit.manager.net;

import com.trtc.uikit.aiconversationkit.store.AIConversationConfig;

public interface AIConversationRequest {
    void startConversation(String roomId, AIConversationConfig config);

    void stopConversation();
}

File: iOS/AIConversationKit/Manager/AIConversationRequest.swift
//
//  AIConversationRequest.swift
//  AIConversationKit
//
//  Created by einhorn on 2025/2/14.
//

import Foundation

class AIConversationRequest {
    
    var config: AIConversationConfig?
    var roomId: String = ""
    
    func start(completion: @escaping (_ taskId: String, _ robotId: String?) -> Void) {
        fatalError("This method must be overridden by subclass")
    }
    
    func stop(taskID: String) {
        fatalError("This method must be overridden by subclass")
    }
    
    func fetchFeedBack() {}
    func fetchExperienceDuration() {}
    func uploadFeedback(_ feedback: [String: String], completion: ((_ code: Int, _ message: String) -> Void)?) {}
}
